app/services/notifications/consolidate_notifications.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module Notifications
+  class ConsolidateNotifications
+    def initialize(from, to, set_data_blk, precondition_blk, threshold)
```

Comment:
I would recommend using keyword arguments here, since there are quite a few parameters and it should make it easier to reason about what is passed in at the call sites.